bridge/client/domutils.js

```diff
@@ -184,11 +185,19 @@ function canConvert(value, expectedTypes) {
         if (type == "object") {
             if (expectedType == "object")
                 return true;
+            if (expectedType == "dictionary")
+                return true;
             // could be a specific object type or host object (e.g. Array)
             var constructor = self[expectedType];
             if (constructor && value instanceof constructor)
                 return true;
         }
+        if (type == "undefined") {
+            if (expectedType == "dictionary") {
+                if (optional)
+                    return true;
+            }
+        }
         if (type == expectedType && expectedType == "function")
```

Comment:
That would work for me. When/where will the actual content of the to-be dictionary be checked if for example setLocal is called with an undefined argument?